src/main/scala/esmeta/error/NotSupported.scala

```diff
@@ -6,4 +6,25 @@ case class NotSupported(reasonPath: NotSupported.ReasonPath)
 object NotSupported:
   type Reason = String
   type ReasonPath = List[Reason]
-  def apply(reason: Reason): NotSupported = NotSupported(List(reason))
+  def unapply(ns: NotSupported): Option[ReasonPath] = Some(ns.reasonPath)
```

Comment:
I think it is unnecessary because already `NotSupported` is defined as a `case class`.
```suggestion
```

Reply:
Ah it was left as this because it was once a normal class while refactoring. Thanks for great catch.

[Delte unnecessary unapply](https://github.com/es-meta/esmeta/pull/126/commits/87f4342c20fb99b463b71a84353f95913b0a49f4) resolvse this.